Automattic/vip-go-mu-plugins
# Search: Fix is_site_indexable() returning incorrect value for blogs with no index

## Description
`get_blog_option()` in `is_site_indexable()` returns `'yes'` for blogs where the `ep_indexable` option doesn't exist:
https://github.com/Automattic/ElasticPress/blob/817e9c7043aa82976a3d2f58710979ee8d382f18/includes/utils.php#L126

This makes sense if you're able to toggle the option by UI, but we've disabled that in:
https://github.com/Automattic/vip-go-mu-plugins/blob/9cf29b6fd38d2fb2b72613d2042190ede324018d/search/includes/classes/class-search.php

Therefore, we should be filtering the `ep_indexable` option depending on whether the index exists or not. 

## Changelog Description

### Plugin Updated: Enterprise Search

Search: Fix is_site_indexable() returning incorrect value for blogs with no index

## Checklist

Please make sure the items below have been covered before requesting a review:


## Steps to Test
1) Set up a multisite and create two sites
2) Index subsite 1 but not 2
3) Shell in and do `\ElasticPress\Utils\is_site_indexable( 2 );` and expect `false`
